File: akka-samples/akka-sample-camel/src/main/scala/Consumer1.scala
package sample.camel

import org.apache.camel.Message

import se.scalablesolutions.akka.util.Logging
import se.scalablesolutions.akka.actor.Actor
import se.scalablesolutions.akka.camel.Consumer

/**
 * @author Martin Krasser
 */
class Consumer1 extends Actor with Consumer with Logging {

  def endpointUri = "file:data/input"

  def receive = {
    case msg: Message => log.info("received %s" format msg.getBody(classOf[String]))
  }

}

File: akka-docs-new/src/main/paradox/java/code/docs/camel/Consumer1.java
package docs.camel;
//#Consumer1
import akka.camel.CamelMessage;
import akka.camel.javaapi.UntypedConsumerActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class Consumer1 extends UntypedConsumerActor {
  LoggingAdapter log = Logging.getLogger(getContext().system(), this);

  public String getEndpointUri() {
    return "file:data/input/actor";
  }

  public void onReceive(Object message) {
    if (message instanceof CamelMessage) {
      CamelMessage camelMessage = (CamelMessage) message;
      String body = camelMessage.getBodyAs(String.class, getCamelContext());
      log.info("Received message: {}", body);
    } else
      unhandled(message);
  }
}
//#Consumer1
